kf_lib_data_ingest/target_apis/kids_first.py

```diff
@@ -269,3 +287,143 @@
+    saved version of the Kids First Data Service schema.
+    :returns output: a dict combining content from Data Service /status
+    endpoint and /swagger endpoint
+    """
+    # Read from file if it exists
+    if os.path.isfile(cached_schema_filepath):
+        return read_json(cached_schema_filepath)
+
+    output = None
+    err = None
+    common_msg = 'Unable to retrieve target schema from target service!'
+    status_url = f'{url}/status'
+    schema_url = f'{url}/swagger'
+
+    # Try to get schemas and version from kf data service
+    try:
+
+        status_response = requests_retry_session().get(status_url)
+        schema_response = requests_retry_session().get(schema_url)
+
+    except ConnectionError as e:
+
+        err = (f'{common_msg}\nCaused by {str(e)}')
+
+    else:
+        if ((status_response.status_code == 200) and
+                (schema_response.status_code == 200)):
+            # API version
+            version = {
+                'version': status_response.json().get('version'),
+                'commit': status_response.json().get('commit')
+            }
+            # Schemas
+            defs = schema_response.json()['definitions']
+            schemas = {
+                k: defs[to_camel_case(k)]
+                for k in target_concepts
+                if to_camel_case(k) in defs
+            }
+            # Add version to output
+            output = {
+                'version': version,
+                'definitions': schemas
+            }
+            # Update cache file
+            write_json(output, cached_schema_filepath)
+        else:
+            err = f'{common_msg}\nCaused by unexpected response(s):'
+            if status_response.status_code != 200:
+                err += f'\nFrom {status_url}:\n{status_response.text}'
+            if schema_response.status_code != 200:
+                err += f'\nFrom {schema_url}/swagger:\n{schema_response.text}'
+
+    # Report error
+    if err:
+        if logger:
+            logger.warning(err)
+        else:
+            print(err)
+
+    return output
+
+
+def handle_nulls(target_instances,
+                 target_api_url=KIDSFIRST_DATASERVICE_PROD_URL):
+    """
+    Convert null property values in `target_instances` to acceptable values
+    based on the type of property as defined in target schema.
+
+    `target_instances` is a dict keyed by the `target_concepts` defined in this
+    module. The values are lists of dicts, where a dict in the list takes on
+    the same form as the dicts in `target_concepts`.
+
+    :param target_instances: a dict of lists containing dicts
+    :param target_schema_url: the endpoint where the target schema will be
+    retrieved
+    :returns target_instances: Updated version of input
+    """
+    # Get the Kids First Data Service schema and version
+    api_schema = _get_kidsfirst_schema(target_api_url)
+    if not api_schema:
+        logger.warning('Skipping null processing')
```

Comment:
```suggestion
        logger.warning('Skipping null processing because target schema not found')
```

Reply:
This looks old ... I may have forgot to push something from last week